src/openrct2/management/Award.cpp

```diff
@@ -606,20 +607,16 @@ void award_update_all()
     {
         // Set active award types as flags
         int32_t activeAwardTypes = 0;
-        int32_t freeAwardEntryIndex = -1;
-        for (int32_t i = 0; i < MAX_AWARDS; i++)
+        for (auto& award : _currentAwards)
         {
-            if (gCurrentAwards[i].Time != 0)
-                activeAwardTypes |= (1 << gCurrentAwards[i].Type);
-            else if (freeAwardEntryIndex == -1)
-                freeAwardEntryIndex = i;
+            activeAwardTypes |= (1 << award.Type);
         }
 
         // Check if there was a free award entry
-        if (freeAwardEntryIndex != -1)
+        if (_currentAwards.size() < MAX_AWARDS)
         {
             // Get a random award type not already active
-            int32_t awardType;
+            uint16_t awardType;
```

Comment:
Why the type change?

Reply:
So that I could use it in a braced initialiser. I'm going to make good first issue to replace all of these uint16_t awardTypes with the enum so thought it would also be good if they were the same type.

Reply:
Ok